### PR TITLE
changes to the type assert code

### PR DESCRIPTION
**Pull request by ardan-bkennedy**

### CODE REVIEW

File: topics/go/language/interfaces/example7/example7.go [hunk 1]
```diff
@@ -21,40 +21,28 @@ func main() {
...
-	case float64:
-		fmt.Println("Passed a float64")
-	default:
-		fmt.Println("Passed something else")
-	}
-
 	switch v := a.(type) {
 	case string:
-		fmt.Println("[string ]: ", v)
+		fmt.Printf("Is string  : type(%T) : value(%s)\n", v, v)
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply topics/go/language/interfaces/example7/example7.go]: Thanks. Yea it helps to show it's real.

File: topics/go/language/interfaces/example7/example7.go [hunk 2]
```diff
@@ -21,40 +21,28 @@ func main() {
...
 
-	// - An interface is satisfied by any type with certain methods.
-	// - The empty interface has no methods.
-	// - All types have AT LEAST no methods.
-	// Therefore all types satisfy the empty interface.
-
-	// - The empty interface says nothing useful for the compiler.
-	// Checks must be performed at runtime to inspect the data
-	// stored in the interface variable.
-	// - Abstracting around the shape of data is costly both in
```
[the remark this answers is not in the document]
Comment (ardan-bkennedy) [reply topics/go/language/interfaces/example7/example7.go]: Let's use the word decoupling and not abstraction. It's a bad word.

IMO, this is not the place for that conversation. This is mechanics. You will teach this when you get to composition.

[the remark this answers is not in the document]
Comment (jcbwlkr) [reply topics/go/language/interfaces/example7/example7.go]: ok I can live with that.

File: topics/go/language/interfaces/example6/example6.go [hunk 3]
```diff
@@ -9,54 +9,54 @@ import (
...
 }
 
+// mockDB defines a mock database we will access.
 type mockDB struct{}
 
-func (db mockDB) findUser(id int) (*user, error) {
-
-	// This is hard coded so code can test against it.
+// find implements the finder interface using pointer semantics.
+func (db mockDB) find(id int) (*user, error) {
```
[the remark this answers is not in the document]
Comment (jcbwlkr) [reply topics/go/language/interfaces/example6/example6.go]: I will think about this. I agree I don't actually mock databases typically. I just want to show a "real world" example of an interface usage here.

**Pull request merged**